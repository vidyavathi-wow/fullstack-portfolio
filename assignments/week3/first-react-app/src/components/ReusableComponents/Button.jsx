import PropTypes from "prop-types";




const Button = ({ label, onClick, type, disabled }) => {
  return (
    <button onClick={onClick} type={type} disabled={disabled}>
      {label}
    </button>
  );
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(["button", "submit", "reset"]),
  disabled: PropTypes.bool,
};


// Button.defaultProps = {
//   type: "button",
//   disabled: false,
//   onClick: () => {},
// };

export default Button;