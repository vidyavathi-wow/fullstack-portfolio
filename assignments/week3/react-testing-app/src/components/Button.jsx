// src/components/Button.jsx
import React from "react";
import PropTypes from "prop-types";

const Button = ({ label, onClick, disabled }) => {
  return (
    <button onClick={onClick} disabled={disabled} data-testid="custom-button">
      {label}
    </button>
  );
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
};

Button.defaultProps = {
  onClick: () => {},
  disabled: false,
};

export default Button;
