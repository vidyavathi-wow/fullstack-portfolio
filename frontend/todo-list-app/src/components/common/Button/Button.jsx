import PropTypes from "prop-types";
import styles from "./Button.module.css";

const Button = ({ children, onClick, type = "button", variant = "primary", disabled = false }) => {
  return (
    <button
      type={type}
      onClick={onClick}
      disabled={disabled}
      className={`${styles.button} ${styles[variant]}`}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(["button", "submit", "reset"]),
  variant: PropTypes.oneOf(["primary", "secondary", "danger"]),
  disabled: PropTypes.bool,
};

export default Button;
