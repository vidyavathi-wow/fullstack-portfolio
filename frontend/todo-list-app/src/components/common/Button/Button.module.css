.button {
  padding: 10px 20px;
  font-size: 16px;
  font-weight: 500;
  border: none;
  border-radius: 6px;
  cursor: pointer;
  transition: background-color 0.3s, color 0.3s, box-shadow 0.2s;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: 6px; 
}

.button:disabled {
  opacity: 0.6;
  cursor: not-allowed;
  box-shadow: none;
}

.primary {
  background-color: #4caf50;
  color: white;
}

.primary:hover:not(:disabled) {
  background-color: #45a049;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.15);
}

.secondary {
  background-color: #e0e0e0;
  color: #333;
}

.secondary:hover:not(:disabled) {
  background-color: #d5d5d5;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.1);
}

/* Danger Variant */
.danger {
  background-color: #f44336;
  color: white;
}

.danger:hover:not(:disabled) {
  background-color: #da190b;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.15);
}
