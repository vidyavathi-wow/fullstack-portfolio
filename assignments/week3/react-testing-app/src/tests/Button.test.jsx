// src/tests/Button.test.jsx
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import Button from "../components/Button.jsx";

describe("Button Component", () => {
  it("renders with the correct label", () => {
    render(<Button label="Click Me" />);
    const button = screen.getByText("Click Me");
    expect(button).toBeInTheDocument();
  });

  it("calls onClick function when clicked", () => {
    const handleClick = vi.fn();
    render(<Button label="Click Me" onClick={handleClick} />);
    
    const button = screen.getByText("Click Me");
    fireEvent.click(button);
    
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it("renders disabled button when disabled prop is true", () => {
    render(<Button label="Can't Click" disabled />);
    const button = screen.getByText("Can't Click");
    expect(button).toBeDisabled();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Button label="Snapshot Test" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
