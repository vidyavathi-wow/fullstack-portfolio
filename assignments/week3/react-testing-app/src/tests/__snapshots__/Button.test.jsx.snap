// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > matches snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="custom-button"
  >
    Snapshot Test
  </button>
</DocumentFragment>
`;
